improver/precipitation_type/shower_condition.py

```diff
@@ -125,33 +120,43 @@ def _output_metadata(self):
         attributes = generate_mandatory_attributes(self.cubes)
         return template, attributes
 
-    def process(self, cloud=None, cloud_texture=None, conv_ratio=None):
+    def process(self, cubes):
         """
         Determine the shower condition from global or UK data depending
-        on input fields
+        on input fields. Expected inputs for UK:
+        cloud_texture: texture_of_low_and_medium_type_cloud_area_fraction,
+        and for global:
+        cloud: low_and_medium_type_cloud_area_fraction
+        conv_ratio: convective_ratio
 
         Args:
-            cloud (iris.cube.Cube or None):
-                Probability of total cloud amount above threshold
-            cloud_texture (iris.cube.Cube or None):
-                Probability of texture of total cloud amount above threshold
-            conv_ratio (iris.cube.Cube or None):
-                Probability of convective ratio above threshold
-
+            cubes (iris.cube.CubeList):
+                List of cubes
         Returns:
             iris.cube.Cube:
                 Binary (0/1) "precipitation is showery"
-
         Raises:
-            ValueError: if inputs are incomplete
+            ValueError: If inputs are incomplete
         """
-        if cloud_texture is None:
-            if cloud is None or conv_ratio is None:
-                raise ValueError("Incomplete inputs")
-            self.cubes = [cloud, conv_ratio]
+
+        cloud = cubes.extract(
+            "probability_of_low_and_medium_type_cloud_area_fraction_above_threshold"
+        )
+        conv_ratio = cubes.extract("probability_of_convective_ratio_above_threshold")
+        cloud_texture = cubes.extract(
+            "probability_of_texture_of_low_and_medium_type_cloud_area_fraction_above_threshold"
+        )
```

Comment:
I was hoping we could achieve something using the information already available in this plugin, so that we don't have to hard-code long strings.  Something like:

```
def probability_name(name):
    return "probability_of_{}_{}_threshold".format(name, name["operator"])

cubes_uk = [cubes.extract(probability_name(name))[0] for name in self.conditions_uk]
cubes_gl = [cubes.extract(probability_name(name))[0] for name in self.conditions_global]

if not np.all(cubes_uk):
    if not np.all(cubes_gl):
        raise ValueError(...)
    self.cubes = cubes_gl
    self.tree = self.conditions_global
else:
    self.cubes = cubes_uk
    self.tree = self.conditions_uk
```
What do you think?

Reply:
This didn't quite work. It took me a while to work out that while all the inputs are _above_ threshold, one operator is _below_ threshold, but this is handled elsewhere in the plugin. I've refactored it a bit more to make sure that exactly one cube is returned from the cubes.extract method.